src/middlewared/middlewared/etc_files/local/smb4.conf

```diff
@@ -0,0 +1,330 @@
+                    pc.update({'winbind use default domain': 'Yes'})
+                pc.update({'winbind refresh tickets': 'Yes'})
+                if db['ad']['ad_nss_info']:
+                    pc.update({'winbind nss info': db['ad']['ad_nss_info']})
+                if not db['ad']['ad_allow_trusted_doms']:
+                    pc.update({'allow trusted domains': 'No'})
+                pc.update({'client ldap sasl wrapping': db['ad']['ad_ldap_sasl_wrapping']})
+                pc.update({'template shell': '/bin/sh'})
+                pc.update({'template homedir': f'{get_cifs_homedir()}/%%D/%%U'})               
+
+            elif db['role'] ==  "ldap_member":
+                pc.update({'server role': 'member server'})
+                pc.update({'security': 'user'}) 
+                if type(db['ldap']['ldap_hostname']) == list:
+                    ldap_pdb_backend = "ldapsam:\"%s\"" & " ".join([
+                        "%s://%s" % ("ldaps" if db['ldap']['ldap_ssl'] == 'on' else "ldap", hostname)
+                        for hostname in db['ldap']['ldap_hostname'].split() 
+                    ])
+                else:
+                    ldap_pdb_backend = f'ldapsam:\"{"ldaps" if db["ldap"]["ldap_ssl"] is not "off" else "ldap"}\"//{db["ldap"]["ldap_hostname"]}'
+                pc.update({'passdb backend': ldap_pdb_backend}) 
+                pc.update({'ldap admin dn': db['ldap']['ldap_binddn']}) 
+                pc.update({'ldap suffix': db['ldap']['ldap_basedn']}) 
+                pc.update({'ldap user suffix': db['ldap']['ldap_user_suffix']}) 
+                pc.update({'ldap group suffix': db['ldap']['ldap_group_suffix']}) 
+                pc.update({'ldap machine suffix': db['ldap']['ldap_machine_suffix']}) 
+                if db['ldap']['ldap_ssl'] is not "start_tls": 
+                    pc.update({'ldap ssl': 'off'}) 
+                pc.update({'ldap replication sleep': '1000'}) 
+                pc.update({'ldap passwd sync': 'Yes'}) 
+                pc.update({'ldapsam trusted': 'Yes'}) 
+                pc.update({'ldap workgroup': db['ldap']['ldap_workgroup']}) 
+                pc.update({'domain logons': 'Yes'})
+
+            elif db['role'] == "domain_controller":
+                pc.update({'server role': 'active directory domain controller'})
+                pc.update({'workgroup': db['DC']['dc_domain'].upper()}) 
+                pc.update({'realm': db['DC']['dc_realm'].upper()})
+                pc.update({'dns forwarder': db['DC']['dc_dns_forwarder']}) 
+                pc.update({'idmap_ldb:use rfc2307': 'yes'})
+
+            elif db['role'] == "standalone":
+                pc.update({'server role': 'standalone'})
+                pc.update({'workgroup': db['cifs']['cifs_srv_workgroup'].upper()})
+
+            return pc
+
+        def add_default_idmap_domain(pc, db):
+            def_idmap = middleware.call_sync('datastore.query', 'directoryservice.idmap_tdb', None, {'get': True})
```

Comment:
get=True would raise an exception if there isnt any value in the db. Are we sure all idmap tables have something in them on this file generation?